config/dependency.go

```diff
@@ -61,7 +65,7 @@ func (targetDepConfig *Dependency) DeepMerge(sourceDepConfig Dependency) error {
 		if targetDepConfig.MockOutputs == nil {
 			targetDepConfig.MockOutputs = sourceDepConfig.MockOutputs
 		} else {
-			newMockOutputs, err := deepMergeCtyMaps(*targetDepConfig.MockOutputs, *sourceDepConfig.MockOutputs)
+			newMockOutputs, err := deepMergeCtyMaps(*targetDepConfig.MockOutputs, *sourceDepConfig.MockOutputs, mergo.WithAppendSlice, mergo.WithOverride)
```

Comment:
I don't like exposing `mergo` at this level. Any preference? Having a `deepMergeCtyMaps` and `deepMergeOverrideCtyMaps` also felt wrong

Reply:
Hmm what's the rationale for having a different deep merge logic for the state merge? I think it would be confusing to have different deep merge behavior across the terragrunt code base.

Reply:
The issue I found with the existing deep merge options (`mergo.WithAppendSlice` and `mergo.WithOverride`) was a `run-all plan` against `fixture-get-output/mock-outputs-merge-strategy-with-state/merge-strategy-with-state-deep` resulted in the following plan for the child module

```
Changes to Outputs:
  ~ test_output1_from_parent               = "value1" -> "fake-output1"
  + test_output2_from_parent               = "fake-abc"
  ~ test_output_list_string                = [
        "a",
      + "fake-list-data",
    ]
  ~ test_output_map_map_string_from_parent = {
      ~ "map_root1"    = {
          ~ "map_root1_sub1" = "map_root1_sub1_value" -> "fake-map_root1_sub1"
        }
      + "not_in_state" = {
          + "abc" = "fake-abc"
        }
    }
```

which is incorrect for a number of reasons

- `test_output_list_string` has the mock appended to the real value
- `test_output_map_map_string_from_parent.map_root.map_root1_sub1` is being replaced from the correct output value to the mock

not using any merge options results in the correct plan

```
Changes to Outputs:
  + test_output2_from_parent               = "fake-abc"
  ~ test_output_map_map_string_from_parent = {
      + "not_in_state" = {
          + "abc" = "fake-abc"
        }
        # (1 unchanged element hidden)
    }
```

Reply:
hi @yorinasub17 , sorry to poke, but do you have any additional thoughts on this? thanks :)

Reply:
@yorinasub17 Updated, thanks for the suggestion

Reply:
Hi @yorinasub17 , sorry to poke again (and the delay. this fell off my radar 😢 )

I've made the changes, are we good to go? thanks